projs/shadow/shadow-engine/core/inc/shadow/core/Thread.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+#include <shadow/exports.h>
+#include <cstddef>
+#include "shadow/util/Syncronization.h"
+
+namespace SH {
+
+  // A thread of execution. Start & manage threads on the fly.
+  struct API Thread {
+
+    explicit Thread();
+    virtual ~Thread();
+
+    // The actual work to do.
+    virtual int Run() = 0;
+
+    // Start the thread and start it running.
+    bool Start(const char* name, bool extended);
```

Comment:
Add proper comments on what the name and extended are used for.

Reply:
The name is the name of the thread. I'm not sure how that can be explained.

Extended exists for something I'm planning later, currently unused.

Reply:
could you add those details as proper doc comments?
Would remove the guess work